File: fixtures/programming/swift/TaskStore.swift
import Foundation

/// A task with title and completion status.
struct Task: Identifiable {
    let id: UUID
    var title: String
    var done: Bool = false

    /// Returns a formatted summary.
    func summary() -> String {
        let status = done ? "✓" : "○"
        return "\(status) \(title)"
    }
}

/// Manages tasks in memory — supports add, complete, and list.
class TaskStore: ObservableObject {
    @Published private(set) var tasks: [Task] = []

    /// Adds a new task with the given title.
    func add(title: String) {
        tasks.append(Task(id: UUID(), title: title))
    }

    /// Marks the task with the given ID as done.
    func complete(id: UUID) {
        guard let index = tasks.firstIndex(where: { $0.id == id }) else { return }
        tasks[index].done = true
    }

    /// Returns all pending tasks.
    var pending: [Task] {
        tasks.filter { !$0.done }
    }
}

File: fixtures/programming/julia/task_store.jl
#= Task management module — uses Julia structs and multiple dispatch. =#

module TaskStore

export Task, TaskCollection, add!, complete!, pending

"""
    Task(id, title; done=false)

A task with an ID, title, and completion status.
"""
struct Task
    id::Int
    title::String
    done::Bool
    Task(id, title; done=false) = new(id, title, done)
end

"""
    TaskCollection()

Mutable collection of tasks with auto-incrementing IDs.
"""
mutable struct TaskCollection
    tasks::Vector{Task}
    next_id::Int
    TaskCollection() = new(Task[], 1)
end

"""Add a task with the given title."""
function add!(col::TaskCollection, title::String)::Task
    task = Task(col.next_id, title)
    push!(col.tasks, task)
    col.next_id += 1
    return task
end

"""Mark a task as done by ID."""
function complete!(col::TaskCollection, id::Int)::Bool
    idx = findfirst(t -> t.id == id, col.tasks)
    idx === nothing && return false
    col.tasks[idx] = Task(id, col.tasks[idx].title; done=true)
    return true
end

"""Return all pending tasks."""
pending(col::TaskCollection) = filter(t -> !t.done, col.tasks)

end # module
